addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1801,6 +1801,225 @@ QUnit.test('composer should be focused automatically after clicking on the send
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    assert.containsNone(
+        document.body,
+        '.o_MessageList',
+        "should not show message list"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_loadingFailed',
+        "should show loading error message"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_loadingFailed .btn',
+        "should show a button to allow user to retry loading"
+    );
+});
+
+QUnit.test('handle error when message fetching failed and there are loaded messages', async function (assert) {
+    assert.expect(4);
+
+    let calls = 0;
+    this.data['mail.channel'].records.push({
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        name: "General",
+    });
+    this.data['mail.message'].records = [...Array(60).keys()].map(id => {
+        return {
+            body: 'coucou',
+            channel_ids: [20],
+            id,
+        };
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            // simulates an error when loading more
+            if (args.method === 'message_fetch') {
+                calls ++;
+                if (calls === 2) {
+                    throw new Error();
+                }
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        30,
+        "should initially show 30 messages"
+    );
+
+    await afterNextRender(() => document.querySelector('.o_MessageList_loadMore').click());
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        30,
+        "should still show 30 messages as load more has failed"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_alertLoadingFailed',
+        "should show loading error message"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_MessageList_loadMore',
+        "should not show load more buttton"
+    );
+});
+
+QUnit.test('retry button shown when message fetching failed triggers load more when there are loaded messages', async function (assert) {
```

Comment:
_"Retry loading more messages on failed load more messages should load more messages"_